tools/image-audit/identity/imageidentity/coloridentity.js

```diff
@@ -0,0 +1,316 @@
+/* eslint-disable class-methods-use-this */
+// import * as ColorThiefModule from 'colorthief';
+import AbstractIdentity from '../abstractidentity.js';
+import IdentityRegistry from '../identityregistry.js';
+import ColorUtility from '../util/colorutility.js';
+import SizeIdentity from './sizeidentity.js';
+
+const ALPHA_ALLOWED_FORMATS = ['png', 'webp', 'gif', 'tiff'];
+
+const numberOfTopColors = 10;
+// const numberOfTopRawColors = 20;
+// eslint-disable-next-line no-undef
+const colorThief = new window.ColorThief();
+
+const exactColorMatchThreshold = numberOfTopColors * 0.2;
+
+const COLOR_IDENTITY_TYPE = 'color-identity';
+
+class ColorIdentity extends AbstractIdentity {
+  #usedColors;
+
+  #topColors;
+
+  constructor(identityState) {
+    super();
+    if (!identityState.usedColors) {
+      identityState.usedColors = new Set();
+      Object.defineProperty(identityState, 'usedColorsSorted', {
+        get() { return ColorUtility.sortColorNamesIntoArray(identityState.usedColors); },
+      });
+    }
+
+    this.#usedColors = identityState.usedColors;
+    this.#topColors = [];
+  }
+
+  get dataObject() {
+    return {
+      topColors: this.#topColors,
+    };
+  }
+
+  static get type() {
+    return COLOR_IDENTITY_TYPE;
+  }
+
+  static get uiSelectorProperties() {
+    return {
+      identity: ColorIdentity.type,
+      display: 'Color',
+      checked: true,
+      hidden: false,
+    };
+  }
+
+  get id() {
+    return 'ci';
+  }
+
+  get strong() {
+    return false;
+  }
+
+  get signleton() {
+    return true;
+  }
```

Comment:
you could theoretically set a default value (i.e. `false`) in the abstract class and let the extending classes overwrite as needed, same as you did for `similarityCollaborator `

Reply:
Good callout. I'll yank this from all the classes and only override for ones that need a true value, for this and the other similar properties.